M/MPICH_CUDA/build_tarballs.jl

```diff
@@ -0,0 +1,158 @@
+
+   # Add /usr/lib/csl-musl-x86_64 to LD_LIBRARY_PATH to be able to use host nvcc
+   export LD_LIBRARY_PATH="/usr/lib/csl-musl-x86_64:/usr/lib/csl-glibc-x86_64:${LD_LIBRARY_PATH}"
+   
+   # Make sure we use host CUDA executable by copying from the x86_64 CUDA redist
+   NVCC_DIR=(/workspace/srcdir/cuda_nvcc-*-archive)
+   rm -rf ${prefix}/cuda/bin
+   cp -r ${NVCC_DIR}/bin ${prefix}/cuda/bin
+   
+   rm -rf ${prefix}/cuda/nvvm/bin
+   cp -r ${NVCC_DIR}/nvvm/bin ${prefix}/cuda/nvvm/bin
+
+   export NVCC_PREPEND_FLAGS="-ccbin='${CXX}'"
+fi
+
+
+export CUDA_HOME=${prefix}/cuda;
+export CUDACXX=$CUDA_HOME/bin/nvcc
+export CUDA_LIB=${CUDA_HOME}/lib
+export CUDA_INCLUDE=${CUDA_HOME}/include
+
+# - Do not install doc and man files which contain files which clashing names on
+#   case-insensitive file systems:
+#   * https://github.com/JuliaPackaging/Yggdrasil/pull/315
+#   * https://github.com/JuliaPackaging/Yggdrasil/issues/6344
+# - `--enable-fast=all,O3` leads to very long compile times for the
+#   file `src/mpi/coll/mpir_coll.c`. It seems we need to avoid
+#   `alwaysinline`.
+
+configure_flags=(
+    --build=${MACHTYPE}
+    --disable-dependency-tracking
+    --disable-doc
+    --enable-fast=ndebug,O3
+    --enable-static=no
+    --host=${target}
+    --prefix=${prefix}
+    --with-device=ch4
+    --with-hwloc=${prefix}
+    --with-cuda-include=${CUDA_INCLUDE}
+    --with-cuda-lib=${CUDA_LIB}
+)
+
+# Define some obscure undocumented variables needed for cross compilation of
+# the Fortran bindings.  See for example
+# * https://stackoverflow.com/q/56759636/2442087
+# * https://github.com/pmodels/mpich/blob/d10400d7a8238dc3c8464184238202ecacfb53c7/doc/installguide/cfile
+export CROSS_F77_SIZEOF_INTEGER=4
+export CROSS_F77_SIZEOF_REAL=4
+export CROSS_F77_SIZEOF_DOUBLE_PRECISION=8
+export CROSS_F77_SIZEOF_LOGICAL=4
+export CROSS_F77_TRUE_VALUE=1
+export CROSS_F77_FALSE_VALUE=0
+
+if [[ ${nbits} == 32 ]]; then
+    export CROSS_F90_ADDRESS_KIND=4
+else
+    export CROSS_F90_ADDRESS_KIND=8
+fi
+export CROSS_F90_OFFSET_KIND=8
+export CROSS_F90_INTEGER_KIND=4
+export CROSS_F90_INTEGER_MODEL=9
+export CROSS_F90_REAL_MODEL=6,37
+export CROSS_F90_DOUBLE_MODEL=15,307
+export CROSS_F90_ALL_INTEGER_MODELS=2,1,4,2,9,4,18,8,
+export CROSS_F90_INTEGER_MODEL_MAP={2,1,1},{4,2,2},{9,4,4},{18,8,8},
+
+./configure "${configure_flags[@]}"
+
+# Remove empty `-l` flags from libtool
+# (Why are they there? They should not be.)
+# Run the command several times to handle multiple (overlapping) occurrences.
+sed -i 's/"-l /"/g;s/ -l / /g;s/-l"/"/g' libtool
+sed -i 's/"-l /"/g;s/ -l / /g;s/-l"/"/g' libtool
+sed -i 's/"-l /"/g;s/ -l / /g;s/-l"/"/g' libtool
+
+# Build the library
+make -j${nproc}
+
+# Install the library
+make install
+
+# Install the license
+install_license $WORKSPACE/srcdir/mpich*/COPYRIGHT
+"""
+
+
+platforms = CUDA.supported_platforms(; min_version = v"11.8")
+platforms = filter!(p -> arch(p) == "x86_64" || arch(p) == "aarch64", platforms)
```

Comment:
Small nitpick: `filter!` modifies its argument and you don't need to assign its result. `filter` does not modify its argument and you need to assign its result.